lib/symbitmatrix.h

```diff
@@ -56,12 +56,14 @@ class SymBitMatrix : private bitvec {
                 if (self.getrange((row*row+row)/2, row+1) != 0) return true;
             } else {
                 if (self.getslice((row*row+row)/2, row+1)) return true; }
-            for (auto c = self.size()-1; c > row; --c)
+            const auto size = self.size();
```

Comment:
do you know why the original iteration was downwards?

Reply:
Honestly, I don't. I guess it was some sort of optimization - the underlying vector allocated just once when it met first non-zero bit. Iterating upwards triggers standard schema of buffer doubling.